### DISCUSSION TITLE
Is there a way to accept external wss feed to socketio?

### DISCUSSION THREAD

**Discussion by weevaa** [miguelgrinberg/Flask-SocketIO#1903] [Q&A]
Miguel, 

Again love your work here thank you for all that you do!  

I have a service that I would like to accept audio data via a wss:// connection to my website and then listen to that stream through the browser.  Is it possible for socketio to handle this scenario?   If so, how would I need to code it to make this work?

**Comment by miguelgrinberg** [comment]
You can stream audio data over Socket.IO messages if you like, but this isn't something that the browser would be able to play on its own. You will also need to build an audio player that receives these chunks of audio and plays them using Audio playback APIs.

If you want to create a stream that the browser automatically recognizes as such and plays, then you have to use standard streaming formats, not Socket.IO or WebSocket.

**Reply by weevaa** [reply]
Thank you so much for the prompt response!  You're awesome!  
How would I allow the external vendor to send the data over socketio by way of wss:// address?  They give me the ability to specify the URL.  

For example would they send it to wss://my-domain.com/socket-io/incoming-audio-stream

If so what would the socketio.on(' ') directive be?  
Something like sockio.on('incoming-audio-stream')?

In your experience said you familiar of any library that would collect and play the chunks as you described?

**Reply by miguelgrinberg** [reply]
This is something you need to arrange with the client that is going to send the audio. Are they going to use Socket.IO or plain WebSocket? You will need to build a server endpoint according to how you agreed the data will be sent.

I'm not familiar with any libraries. The browser has Audio APIs that you can use to play audio buffers.

**Reply by weevaa** [reply]
Plain websocket I believe.  So I guess if have to figure out how to open a separate websocket from socketio and then figure out a way to pipe it over to socketio or just have the browser connect to the secondary wss in order to listen to the audio?

**Reply by miguelgrinberg** [reply]
Socket.IO is not the same as WebSocket. You will not be able to use this library if you intend to receive your audio on a plain WebSocket endpoint. Look for a WebSocket server instead.
